build/yaml/deployBotResources/dotnet/deploy.yml

```diff
@@ -195,7 +195,43 @@ stages:
                 archiveFile: '$(SYSTEM.DEFAULTWORKINGDIRECTORY)/${{ parameters.buildFolder }}/${{ bot.name }}/${{ bot.name }}.zip'
                 replaceExistingArchive: true
                 verbose: true
-          # End of DotNet v3 Install, Build
+          # End of DotNet v3 Install & Build
+
+          # Start of DotNet Composer Install & Build
+          - ${{ if eq(bot.type, 'Composer') }}:
+            # Install dependencies
+            - template: installDependencies.yml
+              parameters:
+                project: "${{ bot.project }}"
+                registry: "$(DEPENDENCIESSOURCE)"
+                version: "$(DEPENDENCIESVERSIONNUMBER)"
+                packages:
+                  Microsoft.Bot.Builder.AI.Luis
+                  Microsoft.Bot.Builder.AI.QnA
+                  Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime
+
+            # Build Bot
+            - task: DotNetCoreCLI@2
+              displayName: "Build"
+              inputs:
+                command: publish
+                publishWebProjects: false
+                projects: "${{ bot.project.directory }}/${{ bot.project.name }}"
+                arguments: "--output $(SYSTEM.DEFAULTWORKINGDIRECTORY)/${{ parameters.buildFolder }}/${{ bot.name }}"
+                modifyOutputPath: false
+
+            # Get BotBuilder package version
+            - task: PowerShell@2
+              displayName: 'Get BotBuilder Version'
+              inputs:
+                targetType: inline
+                workingDirectory: '$(SYSTEM.DEFAULTWORKINGDIRECTORY)/${{ bot.project.directory }}'
+                failOnStderr: true
+                script: |
+                  [XML]$data = Get-Content "./${{ bot.project.name }}"
+                  $package = $data.Project.ItemGroup.PackageReference | Where-Object { $_.Include -eq "Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime" }
+                  Write-Host "##vso[task.setvariable variable=BotBuilderVersionNumber]$($package.version)"
+          # End of DotNet Composer Install & Build
```

Comment:
I think we can reduce all this code and put it in the steps whe its done the same in DotNet Host and Skill.
By in the InstallDependencies condition the script input between Host, Skil do this, and Composer do that. Same for the script in the Get BotBuilder version
![image](https://user-images.githubusercontent.com/62260472/120855322-b2a4c680-c554-11eb-8ee3-9c2a55a24a5f.png)

Reply:
The only step that is the same for composer and the regular dotnet bots is the build, because in the install dependencies it installs different packages, and to get the package version it checks for a different package.
I think getting the steps separate as it is now is more readable.